ui/src/main/webapp/app/gaffer/type-service.js

```diff
@@ -262,10 +262,17 @@ angular.module('app').factory('types', ['config', 'common', function(config, com
         }
 
         if (typeof parts === 'object') {
+            if(type && type["fields"] && type["fields"].length > 0) {
```

Comment:
Can you add some tests for this?

Reply:
I've updated the existing tests, I think it tests the functionality sufficiently - let me know if you need more tests.